scipy/odr/tests/test_odr.py

```diff
@@ -472,3 +474,20 @@ def test_quadratic_model(self):
         output = odr_obj.run()
         assert_array_almost_equal(output.beta, [1.0, 2.0, 3.0])
 
+    def test_output_file_overwrite(self):
+        """
+        Verify fix for gh-1892
+        """
+        def func(b, x):
+            return b[0] + b[1] * x
+
+        p = Model(func)
+        data = Data(np.arange(10), 12 * np.arange(10))
+        ODR(data, p, beta0=[0.1, 13], errfile="error.dat", rptfile='report.dat',
+            ).run()
+        ODR(data, p, beta0=[0.1, 13], errfile="error.dat", rptfile='report.dat',
+            overwrite=True).run()
+        # remove output files for clean up
+        os.remove("error.dat")
```

Comment:
If the test fails, this will leave a stray file behind. Using a `try`-`finally` construct here would be good.

Also, tests may be run on installed `scipy` and it's fairly common for users not to have write access to the install location. So better to use a `tmpfile` location here.

Reply:
I did't care about it, thanks. I fixed it. PTAL.